### DISCUSSION TITLE
getSharedSecret doesn't generate the same secret...

### DISCUSSION THREAD

**Discussion by adavault-ops** [paulmillr/noble-ed25519#48] [Q&A] [answered]
Cannot seem to get the shared secret to be the same using:

`var sharedKeyA = await ed.getSharedSecret(privAlice,pubBob);`

and

`var sharedKeyB = await ed.getSharedSecret(privBob,pubAlice);`

Where

```
        const privAlice = ed.utils.randomPrivateKey(); // 32-byte Uint8Array or string.
        const pubAlice = await ed.getPublicKey(privAlice);
        const privBob = ed.utils.randomPrivateKey(); // 32-byte Uint8Array or string.
        const pubBob = await ed.getPublicKey(privBob);
```

Any clues on why?

**Comment by paulmillr** [comment, marked as answer]
The issue was, as usual, in endianness.

`toX25519()` returned Big endian integer while `scalarMult` consumed Little Endian Uint8Array. I've fixed it in 9f9c2f0. Sorry for that.

**Reply by adavault-ops** [reply]
Cool, will try it out now. Awesome library by the way.
